File: src/main/java/service/BudgetService.java
package service;

import model.Budget;
import model.Transaction;
import model.TransactionType;

public class BudgetService {
    private final Budget budget;

    public BudgetService() {
        this.budget = new Budget();
    }

    public void addIncome(double amount, String description) {
        Transaction income = new Transaction(amount, description, TransactionType.INCOME);
        budget.addTransaction(income);
    }

    public void addExpense(double amount, String description) {
        Transaction expense = new Transaction(amount, description, TransactionType.EXPENSE);
        budget.addTransaction(expense);
    }

    public double getCurrentBalance() {
        return budget.getBalance();
    }

    public void displayTransactions() {
        System.out.println("\nTransaction History:");
        System.out.println("-------------------");
        for (Transaction t : budget.getTransactions()) {
            System.out.printf("%s | %s | $%.2f | %s%n",
                t.getDate(),
                t.getType(),
                t.getAmount(),
                t.getDescription()
            );
        }
        System.out.printf("\nCurrent Balance: $%.2f%n", getCurrentBalance());
    }
}

File: src/services/BudgetService.js
import { Transaction } from '../models/Transaction.js';
import { TransactionType } from '../models/TransactionType.js';
import { Budget } from '../models/Budget.js';
import chalk from 'chalk';

export class BudgetService {
  constructor() {
    this.budget = new Budget();
  }

  addIncome(amount, description) {
    const income = new Transaction(amount, description, TransactionType.INCOME);
    this.budget.addTransaction(income);
  }

  addExpense(amount, description) {
    const expense = new Transaction(amount, description, TransactionType.EXPENSE);
    this.budget.addTransaction(expense);
  }

  getCurrentBalance() {
    return this.budget.getBalance();
  }

  displayTransactions() {
    console.log(chalk.bold('\nTransaction History:'));
    console.log(chalk.gray('-------------------'));
    
    this.budget.getTransactions().forEach(t => {
      const date = t.getDate().toLocaleDateString();
      const type = t.getType() === TransactionType.INCOME 
        ? chalk.green(t.getType()) 
        : chalk.red(t.getType());
      const amount = t.getType() === TransactionType.INCOME
        ? chalk.green(`$${t.getAmount().toFixed(2)}`)
        : chalk.red(`$${t.getAmount().toFixed(2)}`);
      
      console.log(`${date} | ${type} | ${amount} | ${t.getDescription()}`);
    });

    const balance = this.getCurrentBalance();
    console.log(chalk.bold(`\nCurrent Balance: ${
      balance >= 0 
        ? chalk.green(`$${balance.toFixed(2)}`)
        : chalk.red(`$${balance.toFixed(2)}`)
    }`));
  }
}